tests/test_identitystore/test_identitystore.py

```diff
@@ -1,24 +1,607 @@
 """Unit tests for identitystore-supported APIs."""
-from uuid import UUID
+import sys
+from uuid import UUID, uuid4
 
 import boto3
-import sure  # noqa # pylint: disable=unused-import
+import pytest
+from botocore.exceptions import ClientError
+import logging
+import random
+import string
 
 from moto import mock_identitystore
-
+from moto.moto_api._internal import mock_random
 
 # See our Development Tips on writing tests for hints on how to write good tests:
 # http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html
 
+identity_store_id = "d-9067028cf5"
+identity_store_id2 = "d-9067028dg6"
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+
 
 @mock_identitystore
 def test_create_group():
-    client = boto3.client("identitystore", region_name="ap-southeast-1")
-    identity_store_id = "d-9067028cf5"
+    client = boto3.client("identitystore", region_name="us-east-2")
     create_resp = client.create_group(
         IdentityStoreId=identity_store_id,
         DisplayName="test_group",
         Description="description",
     )
     assert create_resp["IdentityStoreId"] == identity_store_id
     assert UUID(create_resp["GroupId"])
+
+
+@mock_identitystore
+def test_create_group_duplicate_name():
+    client = boto3.client("identitystore", region_name="us-east-2")
+    create_resp = client.create_group(
+        IdentityStoreId=identity_store_id,
+        DisplayName="test_group",
+        Description="description",
+    )
+    assert create_resp["IdentityStoreId"] == identity_store_id
+    assert UUID(create_resp["GroupId"])
+
+    try:
```

Comment:
You can use the `with pytest.raises(ClientError)`-construct in all the tests, instead of having to try-except manually: http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html#negative-tests

Reply:
Done. Thanks, that is nice